substrate/client/network/src/litep2p/mod.rs

```diff
@@ -259,6 +266,17 @@ impl Litep2pNetworkBackend {
 		Ok((local_identity, local_peer_id))
 	}
 
+	/// Fetch the number of connected peers from the peerset handle and update
+	/// the atomic `num_connected` shared between the network backend.
+	fn fetch_sync_connected_peers(&self) -> usize {
```

Comment:
Then we would not need this.

Reply:
Have extracted the protocol handler's `num_connected` since that was already under an `Arc<Atomic>`, thanks! 🙏